src/cmd/serial.js

```diff
@@ -233,133 +246,265 @@ module.exports = class SerialCommand extends CLICommandBase {
 	}
 
 	/**
-	 * Check to see if the device is in listening mode, try to get the device ID via serial
+	 * Device identify gives the following
+	 *     - Device ID
+	 *     - (Cellular devices) Cell radio IMEI
+	 *           - Obtained via control req for dvos > 5.6.0
+	 *           - Obtained via serial otherwise
+	 *     - (Cellular devices) Cell radio ICCID
+	 *     - System firmware version
+	 * This command is committed to print the values that are obtained from the device
+	 * ignoring the ones that are not obtained
 	 * @param {Number|String} comPort
 	 */
-	identifyDevice({ port }){
-		let device;
-		return this.whatSerialPortDidYouMean(port, true)
-			.then(_device => {
-				device = _device;
-				if (!device){
-					throw new VError('No serial port identified');
-				}
+	async identifyDevice({ port }) {
+		let deviceFromSerialPort, device;
+
+		let deviceId = '';
+		let fwVer = '';
+		let cellularImei = '';
+		let cellularIccid = '';
+		let isCellular = false;
+
+		// Obtain device
+		try {
+			deviceFromSerialPort = await this.whatSerialPortDidYouMean(port, true);
+			deviceId = deviceFromSerialPort.deviceId;
+			device = await usbUtils.getOneUsbDevice({ idOrName: deviceId });
+		} catch (err) {
+			if (!deviceId) {
+				// The main step has failed and we dont continue
+				// because the others would fail as well
+				throw new VError(err, 'Could not identify device');
+			}
+		}
```

Comment:
Can you extract this `try/catch` sections into small methods? in that way you can have something like:

```
async identifyDevice({port}) {
    ...
   const  device  =  await pickDeviceByPort(port);
   const fw = await getSystemVersion(device);
    ....
}
```